### ISSUE TITLE
[FEATURE] [1.18.2] Requirements for someone to do a Fabric port

### ISSUE THREAD

**Issue by Waterpicker** [Exopandora/ShoulderSurfing#76]
First off, I'm aware you don't want to want to port the mod to fabric yourself. I'm wanting to port itself. I'm asking for any requirements or preferences you have for how a third party port works. You know basic stuff to prevent users from getting confused.

**Comment by Fuzss** [comment]
Hey there, just quickly weighing in on this as I wanted to ask you about helping out with a Fabric port anyways.

[Jared](https://github.com/jaredlll08) has recently demonstrated this incredible [Multiloader Gradle Setup](https://github.com/jaredlll08/MultiLoader-Template) which allows for developing a single mod for multiple mod loaders (even Quilt if you want) at the same time while sharing most of the code. When mod loader specific code is necessary Java's `Service Provider Interface`'s are used. There are no dependencies, the whole thing is pure Gradle, that's what makes it really great. And everything can remain within your own hands, so no need to rely on someone else to maintain a port with feature parity.

For config parity I made [this library](https://www.curseforge.com/minecraft/mc-mods/forge-config-api-port-fabric) which ports the whole Forge config system to Fabric, everything down to the package names is exactly the same. I'm using this in all of my own projects, and some other devs have also begun to make use of this library. It's especially great to use in such a multiloader setup, as the library can already be applied to the common sub project, so only registering the configs is where loader specific code actually has to be used, it's very neat! This would only be a dependency for the Fabric version, nothing on the Forge version would change, still zero dependencies for that one.

I've recently helped someone port their mod to Forge using this setup, feel free to take a look since it shows a lot more applications and features than the template linked above: https://github.com/Fuzss/impactfulweather/tree/multiloader

My own projects also use this setup, although they are much more dependent on my personal setup and have a lot of extra stuff, but still feel free to take a look: [Puzzles Lib](https://github.com/Fuzss/puzzleslib) (my library, that handles a bunch of abstractions so that actual mods have to use loader specific code much more rarely), [Universal Bone Meal](https://github.com/Fuzss/universalbonemeal-forge) (just an example of a mod with some actual content)

If you want to use such a multiloader setup I'd be more than happy to implement it for you and make a PR similar to the Impactful Weather mod linked above, just tell me. 

So to recap, that would mean still no dependencies for Forge, and Fabric would simply depend on Forge Config API Port (and probably Fabric API).

**Comment by IMS212** [comment]
I've ported the entirety of Shoulder Surfing to Fabric as a test, will likely post on GitHub soon.

**Comment by IMS212** [comment]
Done in #79.

**Comment by Exopandora** [comment]
So I finally got some time to work on this and decided to finally port this to fabric (forge will still be supported).

I really like the MultiLoader setup and am currently working to restructure the project for it. The fabric port will most likely use the config library mod @Fuzss mentioned (thanks for the suggestion).

So far, I haven't taken a look at the pull request yet, but i really appreciate the effort you put into it.

**Comment by Exopandora** [comment]
The port to fabric has been done. Thanks for the help and suggestions!
